msger_mosaic_bridge/src/mosaic_adapter.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (C) 2024 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+
+#include "mosaic_adapter.hpp"
+#include "mosaic_client.hpp"
+#include <rapidjson/document.h>
+#include <rapidjson/istreamwrapper.h>
+#include <rapidjson/schema.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+MosaicAdapter::MosaicAdapter() : Node("mosaic_adapter"), mosaic_client_() {
+
+    this->declare_parameter<std::string>("/vehicle_id", "default_vehicle_id");
+    this->declare_parameter<std::string>("role_id", "msg_veh_1");
+    this->declare_parameter<std::string>("cdasim_ip_address", "127.0.0.1");
+    this->declare_parameter<std::string>("host_ip", "127.0.0.1");
+    this->declare_parameter<bool>("enable_registration", true);
+    this->declare_parameter<bool>("enable_vehicle_status", true);
+    this->declare_parameter<int>("registration_port_remote", 6000);
+    this->declare_parameter<int>("registration_port_local", 4001);
+    this->declare_parameter<int>("vehicle_status_port_remote", 7001);
+    this->declare_parameter<int>("vehicle_status_port_local", 4002);
+    this->declare_parameter<int>("siren_and_light_status_port_remote", 8001);
+    this->declare_parameter<int>("siren_and_light_status_port_local", 4003);
```

Comment:
Not sure 100%, but to my interpretation here having both "remote" and "local" ports implies bi-directional communication. Does the vehicle status and siren/light status data get sent bi-directionally?

Reply:
Remove redundant port paramter